### PR TITLE
Add abs, round, floor, ceil operators

### PR DESCRIPTION
**Pull request by anandthakker**
Refs #5853 

TODO: 
 - [x] Round away from 0 instead of towards +Infinity for halfway values. 
 - [x] Add doc entries and SDK support entries to v8.json

### CODE REVIEW

**Comment by anandthakker** [comment]
`"ceil"` matches Javascript's `Math.ceil` -- should we use `"ceiling"` instead?

**Comment by anandthakker** [comment]
cc @samanpwbb

File: src/style-spec/expression/definitions/index.js [hunk 1]
```diff
@@ -309,6 +309,32 @@ CompoundExpression.register(expressions, {
         varargs(NumberType),
         (ctx, args) => Math.max(...args.map(arg => arg.evaluate(ctx)))
     ],
+    'abs': [
+        NumberType,
+        [NumberType],
+        (ctx, args) => Math.abs(args[0].evaluate(ctx))
```
Comment (jfirebaugh) [inline src/style-spec/expression/definitions/index.js]: Prefer argument destructuring as in the other definitions, i.e. `(ctx, [n]) => ...`.

File: src/style-spec/reference/v8.json [hunk 2]
```diff
@@ -2490,6 +2490,42 @@
           }
         }
       },
+      "round": {
+        "doc": "Rounds the input to the nearest integer. Halflway values are rounded away from zero. E.g., `[\"round\", -1.5]` evaluates to -2.",
```
Comment (jfirebaugh) [inline src/style-spec/reference/v8.json]: Halflway → Halfway

Comment (1ec5) [inline src/style-spec/reference/v8.json]: Nit: It’s somewhat unusual to begin a sentence with “E.g.” “For example” is more easily understood.

File: src/style-spec/reference/v8.json [hunk 3]
```diff
@@ -2490,6 +2490,42 @@
...
+        "group": "Math",
+        "sdk-support": {
+          "basic functionality": {
+            "js": "0.45.0"
+          }
+        }
+      },
+      "abs": {
+        "doc": "Returns the absolute value of the input.",
```
Comment (tobrun) [inline src/style-spec/reference/v8.json]: when implementing I noticed abs on `-1.1` returns `1`, could we include in the documentation that abs is an integer type function/not floating point? (from a java background this was a bit confusing for me).

**Comment by anandthakker** [comment]
Oh shoot, that’s a bug! Will fix

**Pull request merged**